src/doc/imagebuf.rst

```diff
@@ -369,57 +369,28 @@ For the remainder of this section, we will assume that you have a
 Example: Visiting all pixels to compute an average color
 --------------------------------------------------------
 
-.. code-block:: cpp
+   .. tabs::
+
+      .. tab:: C++
+            .. literalinclude:: ../../testsuite/docs-examples-cpp/src/docs-examples-imagebuf.cpp
+             :language: c++
+             :start-after: BEGIN-imagebuf-get-pixel-average
+             :end-before: END-imagebuf-get-pixel-average
+             :dedent: 4
 
-    void print_channel_averages (const std::string &filename)
-    {
-        // Set up the ImageBuf and read the file
-        ImageBuf buf (filename);
-        bool ok = buf.read (0, 0, true, TypeDesc::FLOAT);  // Force a float buffer
-        if (! ok)
-            return;
-    
-        // Initialize a vector to contain the running total
-        int nc = buf.nchannels();
-        std::vector<float> total (n, 0.0f);
-    
-        // Iterate over all pixels of the image, summing channels separately
-        for (ImageBuf::ConstIterator<float> it (buf);  ! it.done();  ++it)
-            for (int c = 0;  c < nc;  ++c)
-                total[c] += it[c];
-    
-        // Print the averages
-        imagesize_t npixels = buf.spec().image_pixels();
-        for (int c = 0;  c < nc;  ++c)
-            std::cout << "Channel " << c << " avg = " (total[c] / npixels) << "\n";
-    }
-
-
-.. _sec-make-black:
 
 Example: Set all pixels in a region to black
 --------------------------------------------
 
-.. code-block:: cpp
 
-    bool make_black (ImageBuf &buf, ROI region)
-    {
-        if (buf.spec().format != TypeDesc::FLOAT)
-            return false;    // Assume it's a float buffer
-    
-        // Clamp the region's channel range to the channels in the image
-        roi.chend = std::min (roi.chend, buf.nchannels);
-    
-        // Iterate over all pixels in the region...
-        for (ImageBuf::Iterator<float> it (buf, region);  ! it.done();  ++it) {
-            if (! it.exists())   // Make sure the iterator is pointing
-                continue;        //   to a pixel in the data window
-            for (int c = roi.chbegin;  c < roi.chend;  ++c)
-                it[c] = 0.0f;  // clear the value
-        }
-        return true;
-    }
+   .. tabs::
```

Comment:
This needs to be

```suggestion
   .. tabs:
```

In Sphinx rst, a `::` means "the next indented region is *code*". So if we replace the "`::` followed by indented region" with a `.. tab::` section, we don't want *that* treated as a code example, we want it interpreted as commands.